tests/v1/engine/test_engine_core_client.py

```diff
@@ -87,6 +93,38 @@ async def loop_until_done_async(client: EngineCoreClient, outputs: dict):
             break
 
 
+async def loop_until_fully_done_async(client: EngineCoreClient,
+                                      outputs: dict,
+                                      timeout: float = 20.0):
+    start_time = time.monotonic()
+
+    while True:
+        # Check for timeout
+        elapsed_time = time.monotonic() - start_time
+        if elapsed_time > timeout:
+            raise TimeoutError(
+                f"loop_until_fully_done_async timed out after {timeout} seconds"
+            )
```

Comment:
Instead of the timeout logic, could just call this like:
```pyhton
await asyncio.wait_for(loop_until_fully_done_async(), timeout=20.0)
```

but NBD since it's just a test